lib/ansible/modules/cloud/google/gcp_dns_resource_record_set.py

```diff
@@ -0,0 +1,422 @@
+                                    'dns#resourceRecordSetsListResponse',
+                                    'rrsets')
+    if not result:
+        raise ValueError("Google DNS Managed Zone %s not found" % module.params['managed_zone'])
+    return result
+
+
+def create_change(original, updated, module):
+    auth = GcpSession(module, 'dns')
+    return return_if_change_object(module,
+                                   auth.post(collection(module),
+                                             resource_to_change_request(
+                                                 original, updated, module)
+                                             ))
+
+
+# Fetch current SOA. We need the last SOA so we can increment its serial
+def update_soa(module):
+    original_soa = prefetch_soa_resource(module)
+
+    # Create a clone of the SOA record so we can update it
+    import copy
+    updated_soa = copy.deepcopy(original_soa)
+
+    soa_parts = updated_soa['rrdatas'][0].split(' ')
+    soa_parts[2] = str(int(soa_parts[2]) + 1)
+    updated_soa['rrdatas'][0] = ' '.join(soa_parts)
+    return [original_soa, updated_soa]
+
+
+def resource_to_change_request(original_record, updated_record, module):
+    original_soa, updated_soa = update_soa(module)
+    result = new_change_request()
+    add_additions(result, updated_soa, updated_record)
+    add_deletions(result, original_soa, original_record)
+    return result
+
+
+def add_additions(result, updated_soa, updated_record):
+    if updated_soa:
+        result['additions'].append(updated_soa)
+    if updated_record:
+        result['additions'].append(updated_record)
+
+
+def add_deletions(result, original_soa, original_record):
+    if original_soa:
+        result['deletions'].append(original_soa)
+
+    if original_record:
+        result['deletions'].append(original_record)
+
+
+# TODO(nelsonjr): Merge and delete this code once async operation
+# declared in api.yaml
+def wait_for_change_to_complete(change_id, module):
+    status = 'pending'
+    while status == 'pending':
+        status = get_change_status(change_id, module)
+        if status != 'done':
+            import time
+            time.sleep(0.5)
+
+
+def get_change_status(change_id, module):
+    auth = GcpSession(module, 'dns')
+    link = collection(module, "/%s" % change_id)
+    return return_if_change_object(module, auth.get(link))['status']
+
+
+def new_change_request():
+    import datetime
+    return {
+        'kind': 'dns#change',
+        'additions': [],
+        'deletions': [],
+        'start_time': datetime.datetime.now().isoformat()
+    }
+
+
+def return_if_change_object(module, response):
+    # If not found, return nothing.
+    if response.status_code == 404:
+        return None
+
+    if response.status_code == 204:
+        return None
+
+    try:
+        response.raise_for_status
+        result = response.json()
+    except Exception as inst:
```

Comment:
Instead of global exception, catch JSON errors and requests errors separately, as this block will apply to both.

Reply:
Done.